From 4fcbc34bdb046cefcbe969578e4188061a8db85f Mon Sep 17 00:00:00 2001
From: Aryan Khare <aryakhar@qti.qualcomm.com>
Date: Wed, 15 Oct 2025 15:07:23 +0530
Subject: [PATCH] Fixes PrplMesh Service Status Incorrect Display Also includes
 fix for agent process crash due to unhandled vap node updation

Signed-off-by: Aryan Khare <aryakhar@qti.qualcomm.com>
---
 agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp | 4 ++--
 common/beerocks/scripts/prplmesh_utils.sh.in             | 5 +++--
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp b/agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp
index 749056a30..9a68dcc68 100644
--- a/agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp
+++ b/agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp
@@ -896,8 +896,8 @@ bool Monitor::create_ap_metrics_response(uint16_t mid, const std::vector<sMacAdd
         if (!vap_node) {
             LOG(WARNING) << "Unknown BSSID " << bssid << " - skipping";
 	    update_vaps_in_db();
-	    auto vap_node_refreshed = mon_db.get_vap_node(tlvf::mac_to_string(bssid));
-	    if(!vap_node_refreshed) {
+	    vap_node = mon_db.get_vap_node(tlvf::mac_to_string(bssid));
+	    if(!vap_node) {
 		    LOG(WARNING) << "Unknown BSSID even after updating the vaps in DB again." << bssid << " - skipping";
 		    continue;
 	    }
diff --git a/common/beerocks/scripts/prplmesh_utils.sh.in b/common/beerocks/scripts/prplmesh_utils.sh.in
index 0c740436c..1bfd31ee7 100755
--- a/common/beerocks/scripts/prplmesh_utils.sh.in
+++ b/common/beerocks/scripts/prplmesh_utils.sh.in
@@ -278,9 +278,10 @@ execute_beerocks_command() {
 # Note: Apparently on some Linux version space is added to the process name.
 # Therefore added "($|[[:blank:]])" to the end of the regex expression which means the end of the
 # line ($) or blank character ([[:blank:]])
+
 main_agent_operational() {
-    pgrep -fx "$PRPLMESH_DIR/bin/beerocks_agent($|[[:blank:]])" > /dev/null 2>&1 || return 1
-    grep -q ' --> OPERATIONAL' "$1/beerocks_backhaul@BEEROCKS_LOG_FILES_SUFFIX@"
+    pgrep -fl beerocks_agent  > /dev/null 2>&1 || return 1
+    grep -q ' --> OPERATIONAL' "$1/beerocks_backhaul.log"
 }
 
 radio_agent_operational() {
-- 
2.34.1

